.github/workflows/osc-logs-CI.yml

```diff
@@ -0,0 +1,31 @@
+name: osc-logs-CI
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+
+    - name: osc-logs builds
+      run: go build -v ./main.go
+
+    - name: install jq
+      run: apt-get -y install jq
+
+    - name: Integration tests
+      needs: install jq
+      env:
+        OSC_ACCESS_KEY: ${{ secrets.OSC_ACCESS_KEY }}
+        OSC_SECRET_KEY: ${{ secrets.OSC_SECRET_KEY }}
+        OSC_REGION: ${{ secrets.OSC_REGION }}
+      run: ./tests.sh 
```

Comment:
I suppose it is `./.github/scripts/tests.sh` but you can put `test.sh` content with some few modification in `osc-logs-CI.yml`.
also, you can just rename `osc-logs-CI.yml`to `pull-request.yml`